### PR TITLE
Add Php 8.0 support, bump minimum php 7 to 7.4, update other necessar…

### LINKED ISSUE #1: Support for PHP 8 in a public release
Hi guys, thanks for the great package. 

Is it possible you could make a release with PHP 8 support on your 2.* branch?  (or even 3.0, but I'm not sure about the plans for releasing this).  Just browsing through the code I don't see any reason why the latest release of v2 wouldn't be compatible with PHP 8, but I could be wrong. 

Thanks again.

### PR DESCRIPTION
**Pull request by evang522**
…y dependencies

Resolves #1   

@dragosprotung   just two questions: 

- is the versioning I put in the Changelog Right?

- The newer version of PHPstan which was required in order to update to PHP 8 finds many more errors. I generated a baseline / config file in the dev env, but haven't included it in this commit as I'm not sure how many changes I should bring into the branch. Just let me know if I should add it.   

Thanks :)

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Build
+
+on:
+  push:
+    branches: [ master ]
```
Comment (dragosprotung) [inline .github/workflows/build.yml:5]: Only for the `2.0` branch

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  pull_request:
+    branches: [ master ]
```
Comment (dragosprotung) [inline .github/workflows/build.yml:7]: Only for the `2.0` branch

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,13 @@ All Notable changes to `speicher210/business-hours` will be documented in this f
 
 Updates should follow the [Keep a CHANGELOG](http://keepachangelog.com/) principles.
 
+
+## [2.1.0] - 2021-11-05
```
Comment (dragosprotung) [inline CHANGELOG.md:8]: Do not specify the date yet, simply set it to `2021-XX-XX`

**Review by dragosprotung — CHANGES_REQUESTED** [review]
Let's run the build first so I can see how big the baseline is

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Install lowest dependencies
+        if: ${{ matrix.dependencies == 'lowest' }}
+        run: composer update --prefer-lowest --prefer-dist --no-progress --no-suggest
+
+      - name: Install highest dependencies
+        if: ${{ matrix.dependencies == 'highest' }}
+        run: composer update --prefer-dist --no-progress --no-suggest
+
+      - name: Run checks
+        run: composer check
```
Comment (dragosprotung) [inline .github/workflows/build.yml:49]: You will need to add this to composer script in order to run all the checks. See `master` for example

**Review by dragosprotung — CHANGES_REQUESTED** [review]
I think is OK to create a baseline file and ignore the errors

**Comment by evang522** [comment]
@dragosprotung  
I added the baseline file now, but it has 85 errors in it because of the tests.  

Since in the composer phpstan script the level was different for `tests` (4)  and `src` (6), the tests emitted more errors when set to level 6. Since you can't specify separate levels for different paths in the config, according to https://github.com/phpstan/phpstan/issues/3640 the recommendation is to use the baseline to make this differentiation instead. 

Let me know if I should change it to simply using the separate cli commands though.

**Pull request merged**